src/dma/smoltcp_phy.rs

```diff
@@ -10,8 +10,14 @@ use smoltcp::time::Instant;
 
 /// Use this Ethernet driver with [smoltcp](https://github.com/smoltcp-rs/smoltcp)
 impl<'a, 'rx, 'tx> Device for &'a mut EthernetDMA<'rx, 'tx> {
-    type RxToken<'token> = EthRxToken<'token, 'rx> where Self: 'token;
-    type TxToken<'token> = EthTxToken<'token, 'tx> where Self: 'token;
+    type RxToken<'token>
```

Comment:
Interesting that `cargo fmt` has no opinion on these lines (in the sense that: it accepts either format).

This is definitely better.